pkg/hive/hive.go

```diff
@@ -263,6 +291,23 @@ func (s *Service) checkAndAddPeers(ctx context.Context, peers pb.Peers) {
 	wg := sync.WaitGroup{}
 
 	for _, p := range peers.Peers {
+
+		overlay := swarm.NewAddress(p.Overlay)
+		cacheOverlay := overlay.ByteString()[:cachePrefix]
+
+		// cached peer, skip
+		if _, ok := s.lru.Get(cacheOverlay); ok {
+			continue
+		}
+
+		// mark peer as seen
+		_ = s.lru.Add(cacheOverlay, nil)
```

Comment:
this is suboptimal to do here and better done once the `Ping` returns, since `context` can expire in the meanwhile, we might not even get to the point where we try to ping the peer, resulting in some peers marked as seen when in fact we haven't even checked them yet